test/k8s/egress.go

```diff
@@ -374,6 +374,32 @@ var _ = SkipDescribeIf(func() bool {
 			"loadBalancer.acceleration": "testing-only",
 		},
 	)
+
+	doContext("tunnel vxlan with endpointRoutes enabled and XDP",
```

Comment:
not sure if at this point it makes sense to refactor a bit all the `doContext` calls?

```go
	for _, tunnel := range []string{"disabled", "vxlan"} {
		for _, endpointRoutesEnabled := range []string{"true", "false"} {
			for _, xdpEnabled := range []string{"true", "false"} {
				opts := map[string]string{
					"egressGateway.enabled":     "true",
					"bpf.masquerade":            "true",
					"tunnel":                    tunnel,
					"autoDirectNodeRoutes":      endpointRoutesEnabled,
					"endpointRoutes.enabled":    endpointRoutesEnabled,
					"enableCiliumEndpointSlice": "false",
					"l7Proxy":                   "false",
				}

				if xdpEnabled == "true" {
					opts["loadBalancer.acceleration"] = "testingOnly"
				}

				doContext(fmt.Sprintf("tunnel: %s, endpoint routes: %s, XDP: %s", tunnel, endpointRoutesEnabled, xdpEnabled), opts)
			}
		}
	}
```

Reply:
> not sure if at this point it makes sense to refactor a bit all the `doContext` calls?

That looks prettier indeed :). I think we will soon want to add/switch some configs to use `geneve` instead - will stash this away and see if it still applies then. Thanks!